src/structures/GuildMember.js

```diff
@@ -248,7 +248,9 @@ class GuildMember extends Base {
    */
   edit(data, reason) {
     if (data.channel) {
-      data.channel_id = this.client.channels.resolve(data.channel).id;
+      data.channel = this.client.channels.resolve(data.channel);
+      if (!data.channel) throw new Error('GUILD_CHANNEL_RESOLVE');
```

Comment:
You could also throw the error if the channel is not of type `'voice'`.
```suggestion
      if (!data.channel || data.channel.type !== 'voice') {
        throw new Error('GUILD_CHANNEL_RESOLVE');
      }
```

Reply:
Only thing with this one, is maybe it would then be worth making a new error saying couldn't find a valid *voice* channel instead of a generic guild channel? Though I guess that may be implied that they did something wrong with the channel either way.

Reply:
for example adding `GUILD_VOICE_CHANNEL_RESOLVE: 'Could not resolve channel to a guild voice channel.',` to discord.js/src/errors/Messages.js,

Reply:
I think your suggestion is a bit clearer than mine.